<!-- -*- mode: XML -*- -->
<!-- run velodyne_pointcloud/CloudNodelet in a nodelet manager for a VLP-16 -->

<launch>
  <arg name="veh" default="robot" />

  <group ns="$(arg veh)">
    <group ns="velodyne1">
      <!-- declare arguments with default values -->
      <arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP16db.yaml" />
      <arg name="device_ip" default="192.168.131.202" />
      <arg name="frame_id" default="velodyne1" />
      <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
      <arg name="max_range" default="130.0" />
      <arg name="min_range" default="0.4" />
      <arg name="pcap" default="" />
      <arg name="port" default="2369" />
      <arg name="read_fast" default="false" />
      <arg name="read_once" default="false" />
      <arg name="repeat_delay" default="0.0" />
      <arg name="rpm" default="600.0" />
      <arg name="cut_angle" default="-0.01" />
      <arg name="laserscan_ring" default="-1" />
      <arg name="laserscan_resolution" default="0.007" />

      <!-- start nodelet manager and driver nodelets -->
      <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
        <arg name="device_ip" value="$(arg device_ip)" />
        <arg name="frame_id" value="$(arg frame_id)" />
        <arg name="manager" value="$(arg manager)" />
        <arg name="model" value="VLP16" />
        <arg name="pcap" value="$(arg pcap)" />
        <arg name="port" value="$(arg port)" />
        <arg name="read_fast" value="$(arg read_fast)" />
        <arg name="read_once" value="$(arg read_once)" />
        <arg name="repeat_delay" value="$(arg repeat_delay)" />
        <arg name="rpm" value="$(arg rpm)" />
        <arg name="cut_angle" value="$(arg cut_angle)" />
      </include>

      <!-- start cloud nodelet -->
      <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
        <arg name="calibration" value="$(arg calibration)" />
        <arg name="manager" value="$(arg manager)" />
        <arg name="max_range" value="$(arg max_range)" />
        <arg name="min_range" value="$(arg min_range)" />
      </include>

    </group>

      <node pkg="velodyne_pointcloud" type="lidar_crop" name="lidar_crop" output="screen">
        <param name="x_max" value="0.1" />
        <param name="y_max" value="0.32" />
        <param name="z_max" value="0.3" />
        <param name="x_min" value="0.2" />
        <param name="y_min" value="-0.32" />
        <param name="z_min" value="-1.5" />
        <!-- <param name="~x_ur5_max" value="0.25" />
        <param name="~y_ur5_max" value="0.15" />
        <param name="~z_ur5_max" value="0.1" />
        <param name="~x_ur5_min" value="-0.15" />
        <param name="~y_ur5_min" value="-0.05" />
        <param name="~z_ur5_min" value="-0.2" /> -->
      </node>
    
  </group>
<node name="velodyne_1_to_velodyne_tmp" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 velodyne1 velodyne_tmp" />
</launch>
